server/api/api/endpoints/nuclio.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from fastapi import APIRouter, Depends, Query
+
+import mlrun.common.schemas
+import server.api.utils.clients.async_nuclio
+from server.api.api import deps
+
+router = APIRouter()
+
+
+@router.get("/projects/{project}/nuclio/api-gateways")
+async def list_api_gateways(
+    project: str,
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
+    namespace=Query(default=None, alias="namespace"),
+):
+    await server.api.utils.auth.verifier.AuthVerifier().query_project_permissions(
+        project_name=project,
+        action=mlrun.common.schemas.AuthorizationAction.read,
+        auth_info=auth_info,
+    )
```

Comment:
you have read on project :+1:
next thing - verify read on on project's new entity "api_gateways" as well
see `AuthorizationResourceTypes` where it needs to be added